src/behaviours/__tests__/index.test.tsx

```diff
@@ -754,168 +657,155 @@ describe("dragging", () => {
+
+        expect({
+            x: x.get(),
+            y: y.get(),
+        }).toEqual({
+            x: 42,
+            y: 53,
+        })
 
         // Should update when updating the contstraints again
         render(<Component constraints={{ left: 60, top: 70 }} />)
 
         // Should have updated the x and y
-        expect(x.get()).toEqual(60)
-        expect(y.get()).toEqual(70)
+        expect({
+            x: x.get(),
+            y: y.get(),
+        }).toEqual({
+            x: 60,
+            y: 70,
+        })
     })
 
     test("applies drag transition", async () => {
-        const promise = new Promise(resolve => {
-            const x = motionValue(0)
-            const y = motionValue(0)
-            const Component = () => (
-                <MockDrag>
-                    <motion.div
-                        style={{ x, y }}
-                        drag="x"
-                        dragConstraints={{ left: -500, right: 500 }}
-                        dragElastic
-                        dragMomentum={false}
-                        dragTransition={{
-                            bounceStiffness: 300000,
-                            bounceDamping: 1000000,
-                        }}
-                    />
-                </MockDrag>
-            )
+        const x = motionValue(0)
+        const y = motionValue(0)
+        const Component = () => (
+            <MockDrag>
+                <motion.div
+                    style={{ x, y }}
+                    drag="x"
+                    dragConstraints={{ left: -500, right: 500 }}
+                    dragElastic
+                    dragMomentum={false}
+                    dragTransition={{
+                        bounceStiffness: 300000,
+                        bounceDamping: 1000000,
+                    }}
+                />
+            </MockDrag>
+        )
 
-            const { container, rerender } = render(<Component />)
-            rerender(<Component />)
-
-            const pointer = drag(container.firstChild).to(1, 1)
-            sync.postRender(() => {
-                pointer.to(-500, 0)
-                sync.postRender(() => {
-                    pointer.end()
-                    setTimeout(() => {
-                        // if we set really strong spring and animation has ended - this has worked
-                        resolve(x.get())
-                    }, 50)
-                })
-            })
-        })
+        const { container, rerender } = render(<Component />)
+        rerender(<Component />)
+
+        const pointer = await drag(container.firstChild).to(1, 1)
+        await pointer.to(-500, 0)
+        pointer.end()
+        await sleep(50)
 
-        return expect(promise).resolves.toBeCloseTo(-500)
+        // if we set really strong spring and animation has ended - this has worked
+        expect(x.get()).toBeCloseTo(-500)
     })
 
     test("pointer down kills momentum", async () => {
```

Comment:
Please review change to this test extra carefully - check if this still tests the correct thing